src/odb/include/odb/util.h

```diff
@@ -486,6 +487,61 @@ unsigned int prime)
   }
 };
 
+class RUDYCalculator
+{
+ public:
+  class Tile
+  {
+   public:
+    odb::Rect getRect() { return rect_; }
+    void setRect(int lx, int ly, int ux, int uy);
+    void addRUDY(float rudy);
+    float getRUDY() { return rudy_; }
+
+   private:
+    odb::Rect rect_;
+    float rudy_ = 0;
+  };
+
+  explicit RUDYCalculator(dbBlock* block);
+
+  /**
+   * \pre we need to call this function after `setGridConfig` and
+   * `setWireWidth`.
+   * */
+  void calculateRUDY();
+
+  /**
+   * Set the grid area and grid numbers.
+   * Default value will be the die area of block and (40, 40), respectively.
+   * */
+  void setGridConfig(odb::Rect block, int tileCntX, int tileCntY);
+
+  /**
+   * Set the wire length for calculate RUDY.
+   * If the layer which name is metal1 and it has getWidth value, then this
+   * function will not applied, but it will apply that information.
+   * */
+  void setWireWidth(int wireWidth) { wireWidth_ = wireWidth; }
+
+  Tile& getTile(int x, int y) { return grid_.at(x).at(y); }
```

Comment:
This should be private and a const version should be public to preserve encapsulation.